### PR TITLE
feat: add `string/base/replace-after`

### LINKED ISSUE #812: [RFC]: Add `@stdlib/string/base/replace-after`
### Description

This RFC proposes adding a utility to replace the substring after the first occurrence of a specified search string.

Similar to other base string APIs, the package should assume valid inputs.

Logic should be similar to [`@stdlib/string/substring-after`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/string/substring-after).

Package: `@stdlib/string/base/replace-after`
Alias: `replaceAfter`

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by AuenKr**
Resolves #812 .

## Description

> What is the purpose of this pull request?

This pull request:

-   adding a utility to replace the substring after the first occurrence of a specified search string

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #812 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/string/base/replace-after/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/string/base/replace-after/README.md]:
```suggestion
out = replaceAfter( '', 'xyz', 'foo' );
```

File: lib/node_modules/@stdlib/string/base/replace-after/lib/main.js
Comment (kgryte) [inline lib/node_modules/@stdlib/string/base/replace-after/lib/main.js]:
```suggestion
	return str.substring( 0, idx + search.length ) + replacement;
```

**Comment by kgryte** [comment]
@AuenKr Would you mind updating the copyright years to 2024? Thanks!

**Comment by kgryte** [comment]
@AuenKr Overall, this PR looks great! Thanks for working on this!

**Comment by kgryte** [comment]
@AuenKr There's actually one change to this PR that we should make and that is support for `fromIndex` argument. This was previously discussed in https://github.com/stdlib-js/stdlib/pull/922/files.

The argument should be mandatory (unlike in that discussion), so the expected signature is

```javascript
function replaceAfter( str, search, replacement, fromIndex ) {...}
```

Is this something you can include in this PR?

File: lib/node_modules/@stdlib/string/base/replace-after/lib/main.js
Comment (kgryte) [inline lib/node_modules/@stdlib/string/base/replace-after/lib/main.js]: @AuenKr As mentioned in https://github.com/stdlib-js/stdlib/pull/1363#issuecomment-1962324838, would you mind updating to make `fromIndex` a mandatory argument? Hence, `fromIndex` should not have a default value, but should always be provided by end-users.

The rationale is that this is a "base" package and we try to ensure that base packages are not variadic in order to avoid potential compiler de-optimizations.

File: lib/node_modules/@stdlib/string/base/replace-after/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/string/base/replace-after/README.md]:
```suggestion
Replaces the substring after the first occurrence of a specified search string.
```

File: lib/node_modules/@stdlib/string/base/replace-after/docs/repl.txt
Comment (kgryte) [inline lib/node_modules/@stdlib/string/base/replace-after/docs/repl.txt]:
```suggestion
    fromIndex: integer
```

File: lib/node_modules/@stdlib/string/base/replace-after/docs/repl.txt
Comment (kgryte) [inline lib/node_modules/@stdlib/string/base/replace-after/docs/repl.txt]:
```suggestion
    > out = {{alias}}( 'Hello World!', 'o', 'foo', 5 )
    'Hello Wfoo'
    > out = {{alias}}( 'beep boop beep baz', 'beep', 'foo', 5 )
```

File: lib/node_modules/@stdlib/string/base/replace-after/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/string/base/replace-after/test/test.js]: I don't think this is expected behavior. I would expect that the original input string is returned unchanged.

Comment (kgryte) [reply lib/node_modules/@stdlib/string/base/replace-after/test/test.js]: Providing a `fromIndex` which exceeds the string length should be equivalent to not finding the search string, which in the test which follows correctly confirms that the entire string is returned.

Comment (kgryte) [reply lib/node_modules/@stdlib/string/base/replace-after/test/test.js]: Looking at the implementation, however, the implementation looks correct. Did you confirm that the tests pass?

Ah, wait! You are not testing that `actual` equals `expected` in this test block. That should be updated. I would expect this test case to then fail.

File: lib/node_modules/@stdlib/string/base/replace-after/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/string/base/replace-after/test/test.js:87]: Missing t.strictEqual here and below.

File: lib/node_modules/@stdlib/string/base/replace-after/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/string/base/replace-after/test/test.js]:
```suggestion
tape( 'the function replaces the substring after a provided search string (custom start index)', function test( t ) {
```

**Review by kgryte — CHANGES_REQUESTED** [review]
@AuenKr This looks good. You just need to fix the broken tests, and once finished, this should be ready for merge.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @AuenKr, and thank you for authoring a clean PR! Will merge once CI successfully completes.

**Comment by AuenKr** [comment]
@kgryte, Shouldn't string/base/replace-before also have fromIndex arguement??

**Review by kgryte — APPROVED** [review]

**Comment by kgryte** [comment]
@AuenKr Yes, it should.

**Pull request merged**
